.github/workflows/preview_build.yml

```diff
@@ -0,0 +1,52 @@
+# This action builds and deploys egui_demo_app on each pull request created
+# Security notes:
+# The preview deployment is split in two workflows, preview_build and preview_deploy.
+# `preview_build` runs on pull_request, so it won't have any access to the repositories secrets, so it is safe to
+# build / execute untrusted code.
+# `preview_deploy` has access to the repositories secrets (so it can push to the pr preview repo) but won't run
+# any untrusted code (it will just extract the build artifact and push it to the pages branch where it will
+# automatically be deployed).
```

Comment:
Very nice!

Reply:
…but I guess it doesn't prevent a malicious user from simply changing the `.yml` in the PR.

But there should be some way to configure it so that it is always the `.yml` from `master` that is run? 🤔

Reply:
The pull_request trigger will run workflows from the pr branch, so yes, they could modify the yml. But the workflow won't get any access to secrets and will only get a read only GITHUB_TOKEN, so it wouldn't be a problem. 

Worst thing they could do is change the output to meta.json to point at a different branch and overwrite someone elses preview deployment that way 🤔

It's possible to run the workflow from master via [pull_request_target](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target) but then it would be unsafe to build the untrusted code since workflows using that have access to secrets.